Sources/MapboxCoreNavigation/NavigatorAlternativesStore.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+/**
+ `NavigatorAlternativesStore` observer to track alternative routes updates.
+ */
+public protocol NavigatorAlternativesStoreObserver: AnyObject {
+    /**
+     Called when store has detected a change in alternative routes list.
+     
+     - parameter store: `NavigatorAlternativesStore` reporting an update
+     - parameter didReportNewAlternatives: Indicies within `NavigatorAlternativesStore.alternatives` array where new alternative routes are appended.
+     - parameter removedAlternatives: Array of alternative routes which are no longer actual.
+     */
+    func alternativesStore(_ store: NavigatorAlternativesStore, didReportNewAlternatives: IndexSet, removedAlternatives: [AlternativeRoute])
+    
+    /**
+     Called when store has failed to  change alternative routes list.
+     
+     - parameter store: `NavigatorAlternativesStore` reporting an update
+     - parameter didFailToUpdateAlternatives: An error occured.
+     */
+    func alternativesStore(_ store: NavigatorAlternativesStore, didFailToUpdateAlternatives: AlternativeRouteError)
+}
+
+/**
+ Provides notifications and access to `AlternativeRoute`s found during navigation.
+ */
+public class NavigatorAlternativesStore {
```

Comment:
Why does “Navigator” appear at the beginning of this class name? If it’s because the alternative routes come from the navigator, that’s an internal implementation detail.

“Store” would mean persistent storage of some kind of data, ideally with a [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) interface. That doesn’t seem to describe this class at all. If this class’s only purpose is to keep track of delegates (observers) for alternative route detections, then it would be properly called a “center”, maybe. But since the navigator is an internal implementation detail, we can present a prettier façade to the developer: how about calling this a “detector”, handwaving about the fact that it doesn’t actually do any of the detecting?